### PR TITLE
Fix clang warnings

### PR DESCRIPTION
**Pull request by lysannschlegel**
To get warmed up with the code in the project I had a look at and fixed the warnings clang threw at me.
There's a variety, I recommend you look at the individual commits.

Most of the fixes are straight forward, however some cases of `&& and ||` precedence were not quite clear to me, you should really have a very close look at each of them. I'll mark those I am not sure about.

### CODE REVIEW

File: src/building/construction_clear.c [hunk 1]
```diff
@@ -69,9 +69,9 @@ static int clear_land_confirmed(int measure_only, int x_start, int y_start, int
                 } else if (map_terrain_is(grid_offset, TERRAIN_WATER)) { // keep the "bridge is free" bug from C3
                     continue;
                 } else if (map_terrain_is(grid_offset, TERRAIN_AQUEDUCT) ||
-                           map_terrain_is(grid_offset, TERRAIN_NOT_CLEAR) &&
+                           (map_terrain_is(grid_offset, TERRAIN_NOT_CLEAR) &&
                            !map_terrain_is(grid_offset, TERRAIN_NOT_REMOVABLE) &&
-                           !map_terrain_exists_tile_in_radius_with_type(x, y, 1, 1, TERRAIN_FLOODPLAIN))
+                           !map_terrain_exists_tile_in_radius_with_type(x, y, 1, 1, TERRAIN_FLOODPLAIN)))
```
Comment (lysannschlegel) [inline src/building/construction_clear.c:74]: I wasn't sure about this one.

File: src/figure/action.c [hunk 2]
```diff
@@ -341,7 +341,7 @@ void figure::action_perform() {
             case FIGURE_CART_PUSHER:
                 if (destination_building_id)
                     break;
-                if (!building_is_floodplain_farm(b) && b->state != BUILDING_STATE_VALID || b->figure_id != id)
+                if (!building_is_floodplain_farm(b) && (b->state != BUILDING_STATE_VALID || b->figure_id != id))
```
Comment (lysannschlegel) [inline src/figure/action.c:344]: I wasn't sure about this one.

File: src/figuretype/missile.c [hunk 3]
```diff
@@ -76,8 +76,8 @@ void figure::missile_fire_at(int target_id, int missile_type) {
 
 bool figure::is_citizen() {
     if (action_state != FIGURE_ACTION_149_CORPSE) {
-        if (type && type != FIGURE_EXPLOSION && type != FIGURE_FORT_STANDARD &&
-            type != FIGURE_MAP_FLAG && type != FIGURE_FLOTSAM && type < FIGURE_INDIGENOUS_NATIVE ||
+        if ((type && type != FIGURE_EXPLOSION && type != FIGURE_FORT_STANDARD &&
+             type != FIGURE_MAP_FLAG && type != FIGURE_FLOTSAM && type < FIGURE_INDIGENOUS_NATIVE) ||
```
Comment (lysannschlegel) [inline src/figuretype/missile.c:80]: I wasn't sure about this one.

**Comment by Banderi** [comment]
Ah yeah, I've... been very naughty regarding errors. I would need to take some time and go through them all very carefully, though so far it's low priority, I'm personally still working on refactoring and implementing things first. Good call though!

**Pull request merged**